### PR TITLE
Zierman enhance logging

### PR DESCRIPTION
**Pull request by Zierman**

### CODE REVIEW

File: global_logger.py [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+    VERBOSE = 15
+    DEBUG = logging.DEBUG  # 10
+
+
+CRITICAL = LogLevels.CRITICAL  # 50
+ERROR = LogLevels.ERROR  # 40
+WARNING = LogLevels.WARNING  # 30
+INFO = LogLevels.INFO  # 20
+VERBOSE = LogLevels.VERBOSE # 15
+DEBUG = LogLevels.DEBUG  # 10
```
Comment (Zierman) [inline global_logger.py:29]: I created the container so I could easily loop through all levels in my unit tests. You can still access them as constants directly in the module just like before.

File: global_logger.py [hunk 2]
```diff
@@ -133,24 +145,32 @@ def _wrapper(*args, **kwargs):
...
             else:
                 return return_value
 
-        return classmethod(_wrapper) if self is _LoggedWrapperType.CLASS_METHOD else _wrapper
+        if __wrapped is None:
+            def _outer_wrapper(func):
+                return self.build_wrapper(func, __level)
+            return _outer_wrapper
+        else:
+            return classmethod(_wrapper) if self is _LoggedWrapperType.CLASS_METHOD else _wrapper
```
Comment (Zierman) [inline global_logger.py:170]: This fixes a bug when trying to define a level as a decorator argument.

File: test_global_logger.py [hunk 3]
```diff
@@ -71,23 +69,69 @@ def push(cls, times_pressed=1) -> str:
...
+
+                # check that the line has the correct function
+                list_that_must_be_in_actual = ["CALL A.push",
+                                               "'kwargs': {'times_pressed': 3}",
+                                               "'args': ()",
+                                               "'return_value': 'Someone pushed the button 3 times'"]
+
+                for str_to_find in list_that_must_be_in_actual:
+                    self.assertTrue(str_to_find in added_msg,
+                                    f'{str_to_find!r} was not found in {added_msg!r}')
```
Comment (Zierman) [inline test_global_logger.py:129]: New test to check that using the level keyword on `@logged_class_method` works

File: test_global_logger.py [hunk 4]
```diff
@@ -152,19 +194,64 @@ def __repr__(self):
...
```
Comment (Zierman) [inline test_global_logger.py:254]: New test to check that using the level keyword on `@logged_method` works

File: test_global_logger.py [hunk 5]
```diff
@@ -228,23 +313,65 @@ def push(times_pressed=1) -> str:
...
+
+                # check that the line has the correct function
+                list_that_must_be_in_actual = ["CALL push",
+                                               "'kwargs': {'times_pressed': 3}",
+                                               "'args': ()",
+                                               "'return_value': 'Someone pushed the button 3 times'"]
+
+                for str_to_find in list_that_must_be_in_actual:
+                    self.assertTrue(str_to_find in added_msg,
+                                    f'{str_to_find!r} was not found in {added_msg!r}')
```
Comment (Zierman) [inline test_global_logger.py:372]: New test to check that using the level keyword on `@logged_function` works

**Review by atormanen — APPROVED** [review]

**Pull request merged**